lib/ReactViews/Analytics/DateParameterEditor.tsx

```diff
@@ -15,18 +15,15 @@ const DateParameterEditor: React.FC<DateParameterEditorProps> = observer(
         ? Styles.field
         : Styles.fieldDatePlaceholder;
 
-    const onChangeDate = useCallback(
```

Comment:
What is the type error you are getting for the useCallback, and is the error when using React 18?

Do you think a type annotation in the right place could make the type checker happy? I looked in the React 19 documentation and that seems to still support `useCallback`, and considering this code is from July 2024 (#7228), I'm guessing the decision to use useCallback was made deliberately.

Reply:
There is no type error. It is just better to have it explicitly typed, and this is from the react codemod. There is definitely no reason to have `useCallback` for a function passed to the onClick event. I used to do that in the past, but the only real reason to memoize functions is when passing to use effect or a child component is memoized or building a generic hook so the users don't need to worry about rerendering.